backend/src/openai.ts

```diff
@@ -371,20 +364,13 @@ async function getFinalReplyAfterAllToolCalls(
 	const sentEmails = [];
 	let wonLevel = false;
 
-	const openai = getOpenAI();
 	let gptReply: ChatGptReply | null = null;
 
 	do {
-		gptReply = await chatGptChatCompletion(
-			updatedChatHistory,
-			chatModel,
-			openai
-		);
+		gptReply = await chatGptChatCompletion(updatedChatHistory, chatModel);
```

Comment:
This is inside a do..while loop, so every time it is invoked, we will be creating a new OpenAI instance inside the chatGptChatCompletion function.